loadbalancer/algorithm.go

```diff
@@ -112,19 +114,21 @@ func shiftToRemaining(rnd *rand.Rand, ctx *routing.LBContext, wi []int, now time
 func withFadeIn(rnd *rand.Rand, ctx *routing.LBContext, choice int, algo routing.LBAlgorithm) routing.LBEndpoint {
 	ep := ctx.LBEndpoints
 	now := time.Now()
+	detected := ctx.Registry.GetMetrics(ctx.LBEndpoints[choice].Host).DetectedTime()
 	f := fadeIn(
 		now,
 		ctx.Route.LBFadeInDuration,
 		ctx.Route.LBFadeInExponent,
-		ctx.LBEndpoints[choice].Detected,
+		detected,
 	)
 
 	if rnd.Float64() < f {
 		return ep[choice]
 	}
 	notFadingIndexes := make([]int, 0, len(ep))
 	for i := 0; i < len(ep); i++ {
-		if _, fadingIn := fadeInState(now, ctx.Route.LBFadeInDuration, ep[i].Detected); !fadingIn {
+		detected := ctx.Registry.GetMetrics(ep[i].Host).DetectedTime()
```

Comment:
Detected time never changes but we read it a lot on each request. I am wondering if we can store it in registry put set actual value on the `ep` with a postprocessor.